src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,192 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.utils.CheckedFunction;
+import duke.utils.DukeException;
+import duke.utils.Record;
+
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class Parser {
+
+    private final TaskList db;
+    private final HashMap<String, CheckedFunction<String, Record>> cmds = new HashMap<>();
+
+    /**
+     * Creates a Parser which initializes a TaskList.
+     *
+     * @param loadFromStorage Inform the TaskList whether to load tasks from storage.
+     * @throws DukeException
+     */
+    public Parser(boolean loadFromStorage) throws DukeException {
+        this(new TaskList(loadFromStorage));
+    }
+
+    /**
+     * Creates a Parser with TaskList db and with the recognized commands.
+     *
+     * @param db TaskList to be added.
+     * @throws DukeException
+     */
+    public Parser(TaskList db) throws DukeException {
+        this.db = db;
+        cmds.put("greet", this::greet);
+        cmds.put("bye", this::bye);
+        cmds.put("list", this::list);
+        cmds.put("done", this::done);
+        cmds.put("delete", this::delete);
+        cmds.put("todo", this::todo);
+        cmds.put("deadline", this::deadline);
+        cmds.put("event", this::event);
+        cmds.put("help", this::help);
+        cmds.put("clear", this::clear);
+        cmds.put("find", this::find);
+    }
+
+    /**
+     * Takes in a command, and if command is found, passes to it the remaining arguments.
+     *
+     * @param input
+     * @return
+     * @throws DukeException
+     */
+    public Record query(String input) throws DukeException {
+        try {
+            Scanner sc = new Scanner(input);
+            String cmd = sc.next();
+            String raw = sc.hasNext() ? sc.nextLine().substring(1) : "";
+            if (cmds.containsKey(cmd)) {
+                return cmds.get(cmd).apply(raw);
+            } else {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (NoSuchElementException e) {
+            throw new DukeException("Please enter a command.");
+        }
+    }
+
+    private String sizeMsg() {
```

Comment:
Perhaps you could use a verb for this method name? Something like `getTaskSize()`